go/src/lib.rs

```diff
@@ -639,3 +645,181 @@ fn get_slot_addr(slot_type: &protobuf::EnumOrUnknown<SlotTypes>) -> SlotAddr {
         })
         .expect("Received unexpected slot id type")
 }
+
+// This struct is used to keep track of the cursor of a cluster scan.
+#[repr(C)]
+pub struct ClusterScanCursor {
+    cursor: *const c_char,
+}
+
+impl ClusterScanCursor {
+    #[no_mangle]
+    pub extern "C" fn new_cluster_cursor(new_cursor: *const c_char) -> Self {
```

Comment:
Need to be marked as `unsafe` with the corresponding docs
I think @jonathanl-bq can tell you more